src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/ILGen.cs

```diff
@@ -947,6 +947,10 @@ private static void EmitNumericConversion(this ILGenerator il, Type typeFrom, Ty
                                 il.Emit(OpCodes.Conv_I8);
                             }
                             break;
+                        case TypeCode.Boolean:
+                            il.Emit(OpCodes.Ldc_I4_0);
```

Comment:
Both should work, to pick between the two this is closer to comparable code I found in writing a bunch of C# code with boolean results and seeing what the compiler did with it, so I went with that.
I'm also not sure what would happen in that case if someone weirdly forced through a value other than 1 or 0, but know this would work so didn't try. (I once coerced numbers to bools as a shortcut in CIL, and my unit tests treated them as Schrödinger's booleans).